# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: startupcache/StartupCache.h
# Commit: 7b7b147b6955
# Full Hash: 7b7b147b6955cee07e0c115993446bfbd59cf7e2
# Author: Doug Thayer <dothayer@mozilla.com>
# Date: 2020-02-13 03:57:45
# Regressor Bug: 1607505
# File Overlap Count: 3
# Description:
#   Bug 1607505 - Start writing StartupCache earlier in shutdown r=froydnj
#   
#   The initial thought for getting the StartupCache out of the shutdown
#   path was to simply not write it during shutdown, as it should write
#   60 seconds after startup, and the theory was that if the user shut
# ==============================================================================

diff -r 0b8589ddbacf -r 7b7b147b6955 startupcache/StartupCache.h
--- a/startupcache/StartupCache.h	Wed Feb 12 16:22:10 2020 +0000
+++ b/startupcache/StartupCache.h	Wed Feb 12 19:02:12 2020 +0000
@@ -163,6 +163,10 @@
   // Removes the cache file.
   void InvalidateCache(bool memoryOnly = false);
 
+  // For use during shutdown - this will write the startupcache's data
+  // to disk if the timer hasn't already gone off.
+  void MaybeInitShutdownWrite();
+
   // Signal that data should not be loaded from the cache file
   static void IgnoreDiskCache();
 
@@ -203,6 +207,7 @@
   void WaitOnWriteThread();
   void WaitOnPrefetchThread();
   void StartPrefetchMemoryThread();
+  void MaybeSpawnWriteThread();
 
   static nsresult InitSingleton();
   static void WriteTimeout(nsITimer* aTimer, void* aClosure);